.github/workflows/build-and-test.yml

```diff
@@ -219,3 +219,89 @@ jobs:
       - name: Build python binding
         run: cargo build --release
         working-directory: bindings/python/native
+
+  java-bindings:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install LLVM and Clang # required for bindgen to work, see https://github.com/rust-lang/rust-bindgen/issues/1797
+        uses: KyleMayes/install-llvm-action@32c4866ebb71e0949e8833eb49beeebed48532bd
+        if: matrix.os == 'windows-latest'
+        with:
+          version: "11.0"
+          directory: ${{ runner.temp }}/llvm
+
+      - name: Set LIBCLANG_PATH
+        run: echo "LIBCLANG_PATH=$((gcm clang).source -replace "clang.exe")" >> $env:GITHUB_ENV
+        if: matrix.os == 'windows-latest'
+
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+
+      - name: Set up Java 11
+        uses: actions/setup-java@v1
+        with:
+            java-version: 11.0.x
+            architecture: x64
+
+      - name: Get current date
+        if: matrix.os == 'windows-latest'
+        run: echo "CURRENT_DATE=$(Get-Date -Format "yyyy-MM-dd")" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+
+      - name: Get current date
+        run: echo "CURRENT_DATE=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+        if: matrix.os == 'macos-latest' || matrix.os == 'ubuntu-latest'
+
+      - name: Get current date
+        if: matrix.os == 'windows-latest'
+        run: echo "CURRENT_DATE=$(Get-Date -Format "yyyy-MM-dd")" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
```

Comment:
This looks like a duplicate of lines 254-256

Reply:
@rajivshah3 Thanks for checking this one, im not sure what those were exactly doing but they exist in python and nodejs as well 🙂  105-109 and 173-178

Reply:
It gets the current date and uses that as part of the cache key for the registry, index, and target caches

Reply:
Whats your opinion on java 11 testing? could do 13? or 8 :P I tried to get another matrix working, but i didnt manage, and also unsure if thats entirely needed

Reply:
Yeah that can be done if you want. You can specify the versions here:

https://github.com/iotaledger/wallet.rs/blob/21099e5a0a6bbcd2e266f631d7af72d6b9c41f1a/.github/workflows/covector-version-or-publish.yml#L56

and then pass the version to `setup-java` like this:
https://github.com/iotaledger/wallet.rs/blob/21099e5a0a6bbcd2e266f631d7af72d6b9c41f1a/.github/workflows/covector-version-or-publish.yml#L66

Reply:
It looks like you got it to work in 9d8bc8ec2642fbe5cb2cc091e2e49317ae1e3bec but then ran into the issue with Clang on Windows

Reply:
In https://github.com/iotaledger/wallet.rs/commit/7e81ffa61a3eddab22c4cff0b77aeb32e715c5ec it should work, you just need to add `java` to the matrix under the `java-bindings` job